tools/build.py

```diff
@@ -230,6 +220,27 @@ def adjust_options(options):
     options.host_tuple = '%s-%s' % (platform.arch(), platform.os())
     options.target_tuple = '%s-%s' % (options.target_arch, options.target_os)
 
+    # Verify options consistency (sysroot vs. cross build)
+    if options.target_tuple != options.host_tuple:
+        # This is cross build, sysroot should/must be supplied
+        if options.target_os in ['nuttx', 'tizenrt']:
+            require_sysroot = True
+            options.buildlib = True
+        if not options.sysroot:
+            if require_sysroot:
+                ex.fail('--sysroot required for nuttx/tinyara')
+            else:
+                ex.warn('--sysroot recommended for cross builds target')
+        options.sysroot = fs.abspath(options.sysroot)
+        if not fs.exists(options.sysroot):
+            ex.fail('sysroot %s does not exist' % options.sysroot)
+
+    arch_for_iotjs = 'arm' if options.target_arch[0:3] == 'arm' else \
```

Comment:
`options.target_arch.startswith('arm')` might be better. No need to juggle with substring indices.